Generate bb bw and vcf files when there are no reads in the bam file

cadley-nyulangone opened the pull request.

mattmaurano commented on dnase/analysis.sh:
```diff
@@ -439,7 +439,6 @@ elif [[ "${sampleType}" == "dnase" ]] || [[ "${sampleType}" == "atac" ]] || [[ "
         fi
         #Kent tools can't use STDIN
         bedGraphToBigWig $TMPDIR/${name}.${mappedgenome}.perBase.clipped.bedGraph ${chromsizes} ${sampleOutdir}/${name}.${mappedgenome}.perBase.bw
-        
```
stray line

cadley-nyulangone replied:
Not sure why line 442 looks that way on the right, in green.  There are only 2 tabs in the code.

mattmaurano replied:
it looks like you added whitespace -- you're saying that's not the case?

mattmaurano commented on dnase/callsnpsMerge.sh:
```diff
@@ -72,10 +72,15 @@ date
-rm -f ${covfiles}
 
-unstarch ${sampleOutdir}/${name}.${mappedgenome}.coverage.starch | cut -f1-3,5 > $TMPDIR/${name}.${mappedgenome}.coverage.bedGraph
+if [ "${covfiles}" = "" ]; then
```
why would this be empty? doesn't your change above ensure inputs.callsnps has entries?

cadley-nyulangone commented:
Yes, you're right.  I pulled out all those if blocks from callsnpsMerge.sh
  Bad weekend.

On Mon, Dec 14, 2020 at 9:49 AM mattmaurano <notifications@github.com>
wrote:

> *@mattmaurano* requested changes on this pull request.
> ------------------------------
>
> In dnase/analysis.sh
> <https://github.com/mauranolab/mapping/pull/243#discussion_r542440961>:
>
> > @@ -439,7 +439,6 @@ elif [[ "${sampleType}" == "dnase" ]] || [[ "${sampleType}" == "atac" ]] || [[ "
>          fi
>          #Kent tools can't use STDIN
>          bedGraphToBigWig $TMPDIR/${name}.${mappedgenome}.perBase.clipped.bedGraph ${chromsizes} ${sampleOutdir}/${name}.${mappedgenome}.perBase.bw
> -
>
> stray line
> ------------------------------
>
> In dnase/callsnpsMerge.sh
> <https://github.com/mauranolab/mapping/pull/243#discussion_r542441827>:
>
> >
> -unstarch ${sampleOutdir}/${name}.${mappedgenome}.coverage.starch | cut -f1-3,5 > $TMPDIR/${name}.${mappedgenome}.coverage.bedGraph
> +if [ "${covfiles}" = "" ]; then
>
> why would this be empty? doesn't your change above ensure inputs.callsnps
> has entries?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mauranolab/mapping/pull/243#pullrequestreview-551439278>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY4NJS3EU73MK6UIOEMZD3SUYQXRANCNFSM4U22QW3Q>
> .
>

mattmaurano commented:
No worries. Can you double check by running a good and bad sample from the latest FC?

cadley-nyulangone commented:
I had added tabs, but it wanted whitespace.  Fixed now.

On Mon, Dec 14, 2020 at 11:13 AM mattmaurano <notifications@github.com>
wrote:

> *@mattmaurano* commented on this pull request.
> ------------------------------
>
> In dnase/analysis.sh
> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>:
>
> > @@ -439,7 +439,6 @@ elif [[ "${sampleType}" == "dnase" ]] || [[ "${sampleType}" == "atac" ]] || [[ "
>          fi
>          #Kent tools can't use STDIN
>          bedGraphToBigWig $TMPDIR/${name}.${mappedgenome}.perBase.clipped.bedGraph ${chromsizes} ${sampleOutdir}/${name}.${mappedgenome}.perBase.bw
> -
>
> it looks like you added whitespace -- you're saying that's not the case?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY4NJWSZT4L6SBLZ4OLIWTSUY2TTANCNFSM4U22QW3Q>
> .
>

cadley-nyulangone commented:
OK

On Mon, Dec 14, 2020 at 11:17 AM John Cadley <cadley.nyulangone@gmail.com>
wrote:

> I had added tabs, but it wanted whitespace.  Fixed now.
>
> On Mon, Dec 14, 2020 at 11:13 AM mattmaurano <notifications@github.com>
> wrote:
>
>> *@mattmaurano* commented on this pull request.
>> ------------------------------
>>
>> In dnase/analysis.sh
>> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>:
>>
>> > @@ -439,7 +439,6 @@ elif [[ "${sampleType}" == "dnase" ]] || [[ "${sampleType}" == "atac" ]] || [[ "
>>          fi
>>          #Kent tools can't use STDIN
>>          bedGraphToBigWig $TMPDIR/${name}.${mappedgenome}.perBase.clipped.bedGraph ${chromsizes} ${sampleOutdir}/${name}.${mappedgenome}.perBase.bw
>> -
>>
>> it looks like you added whitespace -- you're saying that's not the case?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AKY4NJWSZT4L6SBLZ4OLIWTSUY2TTANCNFSM4U22QW3Q>
>> .
>>
>

cadley-nyulangone commented:
Ran *BS08612A.cegsvectors_LP300* (good) and
*BS08612A.cegsvectors_pSpCas9GFP* (bad).  Both working now.

On Mon, Dec 14, 2020 at 11:18 AM John Cadley <cadley.nyulangone@gmail.com>
wrote:

> OK
>
> On Mon, Dec 14, 2020 at 11:17 AM John Cadley <cadley.nyulangone@gmail.com>
> wrote:
>
>> I had added tabs, but it wanted whitespace.  Fixed now.
>>
>> On Mon, Dec 14, 2020 at 11:13 AM mattmaurano <notifications@github.com>
>> wrote:
>>
>>> *@mattmaurano* commented on this pull request.
>>> ------------------------------
>>>
>>> In dnase/analysis.sh
>>> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>:
>>>
>>> > @@ -439,7 +439,6 @@ elif [[ "${sampleType}" == "dnase" ]] || [[ "${sampleType}" == "atac" ]] || [[ "
>>>          fi
>>>          #Kent tools can't use STDIN
>>>          bedGraphToBigWig $TMPDIR/${name}.${mappedgenome}.perBase.clipped.bedGraph ${chromsizes} ${sampleOutdir}/${name}.${mappedgenome}.perBase.bw
>>> -
>>>
>>> it looks like you added whitespace -- you're saying that's not the case?
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/mauranolab/mapping/pull/243#discussion_r542510195>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AKY4NJWSZT4L6SBLZ4OLIWTSUY2TTANCNFSM4U22QW3Q>
>>> .
>>>
>>
